Shared/mods/deathmatch/logic/luadefs/CLuaCryptDefs.cpp

```diff
@@ -614,10 +765,17 @@ int CLuaCryptDefs::DecodeString(lua_State* luaVM)
     CScriptArgReader argStream(luaVM);
     argStream.ReadEnumString(algorithm);
     argStream.ReadString(data);
-    argStream.ReadStringMap(options);
 
-    argStream.ReadFunction(luaFunctionRef, LUA_REFNIL);
-    argStream.ReadFunctionComplete();
+    if ((algorithm != StringEncodeFunction::BASE64 && algorithm != StringEncodeFunction::BASE32) || argStream.NextIsTable())
+    {
+        argStream.ReadStringMap(options);
+    }
+
+    if ((algorithm != StringEncodeFunction::BASE64 && algorithm != StringEncodeFunction::BASE32) || argStream.NextIsFunction())
```

Comment:
This forces the use of async `function` for the existing algorithms while omit the requirement for the new ones. This may also confuse and break the backward compatibility.

Reply:
Although, this condition makes no sense because of the `LUA_REFNIL` argument.

Reply:
Thank you, done.